src/api/model/modelApi.test.ts

```diff
@@ -14,133 +14,79 @@
-  ];
-  const database = 'test-db';
-  const engine = 'test-engine';
-
-  afterEach(() => nock.cleanAll());
-  afterAll(() => nock.restore());
-
-  it('should install model', async () => {
-    const scope = nockTransaction(
-      [
-        {
-          type: 'InstallAction',
-          sources: mockModels,
-        },
-      ],
-      [
-        {
-          type: 'InstallActionResult',
-        },
-      ],
-      database,
-      engine,
-      false,
-    );
-    const result = await api.installModels(database, engine, mockModels);
-
-    scope.done();
+import Client from '../client';
+import { Model } from '../transaction/types';
 
-    expect(result).toEqual(
-      makeTransactionResult([
-        {
-          type: 'InstallActionResult',
-        },
-      ]),
-    );
-  });
+describe('Integration', () => {
+  const databaseName = `js-sdk-tests-${Date.now()}`;
+  const engineName = `js-sdk-tests-${Date.now()}`;
+  let client: Client;
 
-  it('should list models', async () => {
-    const scope = nockTransaction(
-      [
-        {
-          type: 'ListSourceAction',
-        },
-      ],
-      [
-        {
-          type: 'ListSourceActionResult',
-          sources: mockModels,
-        },
-      ],
-      database,
-      engine,
-    );
-    const result = await api.listModels(database, engine);
+  jest.setTimeout(1000 * 60 * 10);
 
-    scope.done();
+  beforeAll(async () => {
+    client = await getClient();
 
-    expect(result).toEqual(
-      makeTransactionResult([
-        {
-          type: 'ListSourceActionResult',
-          sources: mockModels,
-        },
-      ]),
-    );
+    await createEngineIfNotExists(client, engineName);
+    await createDatabaseIfNotExists(client, databaseName);
   });
 
-  it('should get model', async () => {
-    const scope = nockTransaction(
-      [
-        {
-          type: 'ListSourceAction',
-        },
-      ],
-      [
-        {
-          type: 'ListSourceActionResult',
-          sources: mockModels,
-        },
-      ],
-      database,
-      engine,
-    );
-    const result = await api.getModel(database, engine, 'model2');
-
-    scope.done();
-
-    expect(result).toEqual(mockModels[1]);
+  afterAll(async () => {
+    await client.deleteEngine(engineName);
+    await client.deleteDatabase(databaseName);
   });
 
-  it('should delete model', async () => {
-    const scope = nockTransaction(
-      [
-        {
-          type: 'ModifyWorkspaceAction',
-          delete_source: ['model1'],
-        },
-      ],
-      [
-        {
-          type: 'ModifyWorkspaceActionResult',
-        },
-      ],
-      database,
-      engine,
-      false,
-    );
-    const result = await api.deleteModel(database, engine, 'model1');
+  describe('Models actions', () => {
+    it('should listModels', async () => {
+      const models = await client.listModels(databaseName, engineName);
+      expect(models?.length).toBeGreaterThan(0);
```

Comment:
The return type is inferred as `Promise<any[] | undefined>`. That's not good.

![image](https://user-images.githubusercontent.com/6468871/192589469-e7f11cf5-bede-4a8a-987c-22beea77c11e.png)

It should be `Promise<string[]>`